app/Scripts/Stats.gd

```diff
@@ -1,27 +1,58 @@
 extends Node
-signal healthChange
-signal staminaChange
-var Health setget ChangedHealth
-var staminaForDash = 10
-var MaxHealth = 100
+#_____________________#
+
+signal Death
+signal healthChange(value)
+signal staminaChange(value)
+signal boidsChange(value)
+#_____________________#
+
+var BoidsCollectedNum setget BoidsChanged
 var Stamina setget ChangedStamina
+var Health = 10 setget ChangedHealth
+var FollowingBoids = []
+
+#_____________________#
+var staminaForDash = 10
+var BoidHealthNum = 0
+
 var MaxStamina = 100
 var staminaRegen = 5
+export var HealthPerBoid = 10
+
+#_____________________#
+
+func BoidsChanged(value): # change the value of the number of boids you get
+	
+	BoidHealthNum = value - BoidsCollectedNum
+	BoidsCollectedNum = value
```

Comment:
I think the number of boids should be derived? Like we have a collection of boids, the size of that collection is always known... We can use what you have for now but something to think about

Reply:
also is this function used anywhere? If it's not we shouldn't be adding it in this PR. But maybe it's used somewhere in the editor or something it's not visible in the git changes?

Reply:
Its used in the Hud

Reply:
yep you are right, Idk why I didn't think of that lol